src/v3/recordingComputeUtils.test.ts

```diff
@@ -237,6 +240,431 @@ describe('recordingComputeUtils', () => {
+            createMockSpanAndAnnotation(200, {
+              name: 'end-span',
+              duration: 50,
+            }),
+            createMockSpanAndAnnotation(400, {
+              // matching span
+              name: 'end-span',
+              duration: 50,
+            }),
+            createMockSpanAndAnnotation(600, {
+              name: 'end-span',
+              duration: 50,
+            }),
+            createMockSpanAndAnnotation(700, {
+              name: 'end-span',
+              duration: 50,
+            }),
+          ]),
+          input: {
+            id: 'test',
+            startTime: createTimestamp(0),
+            relatedTo: {},
+            variant: 'origin',
+          },
+          recordedItemsByLabel: {},
+        })
+        expect(result['test-computed-span']).toEqual({
+          duration: 650, // (700 + 50) - 100
+          startOffset: 100,
+        })
+      })
+
+      it('should not return any computed spans using a invalid matchingIndex', () => {
+        const definition: CompleteTraceDefinition<
+          'global',
+          AnyRelation,
+          'origin'
+        > = {
+          ...baseDefinitionFixture,
+          computedSpanDefinitions: {
+            'test-computed-span': {
+              startSpan: ({ span }) => span.name === 'start-span',
+              endSpan: Object.assign(
+                ({ span }: { span: Span<AnyRelation> }) =>
+                  span.name === 'end-span',
+                {
+                  matchingIndex: -100,
+                },
+              ),
+            },
+          },
+        }
+        const result = getComputedSpans({
+          definition,
+          recordedItems: new Set([
+            createMockSpanAndAnnotation(100, { name: 'start-span' }),
+            createMockSpanAndAnnotation(200, {
+              name: 'end-span',
+              duration: 50,
+            }),
+            createMockSpanAndAnnotation(400, {
+              // matching span
+              name: 'end-span',
+              duration: 50,
+            }),
+            createMockSpanAndAnnotation(600, {
+              name: 'end-span',
+              duration: 50,
+            }),
+            createMockSpanAndAnnotation(700, {
+              name: 'end-span',
+              duration: 50,
+            }),
+          ]),
+          input: {
+            id: 'test',
+            startTime: createTimestamp(0),
+            relatedTo: {},
+            variant: 'origin',
+          },
+          recordedItemsByLabel: {},
+        })
+        expect(result).toEqual({})
+      })
+
+      it('should work with span definition objects containing matchingIndex', () => {
+        const definition: CompleteTraceDefinition<
+          'global',
+          AnyRelation,
+          'origin'
+        > = {
+          ...baseDefinitionFixture,
+          computedSpanDefinitions: {
+            'test-computed-span': {
+              startSpan: { name: 'start-span', matchingIndex: 1 },
```

Comment:
like the typescript failure says, this needs to be wrapped in `fromDefinition(...)`